crypto/types/src/types.rs

```diff
@@ -643,61 +666,55 @@ pub type PacketTag = [u8; PACKET_TAG_LENGTH];
 
 /// Represents a secp256k1 public key.
 /// This is a "Schrödinger public key", both compressed and uncompressed to save some cycles.
+///
+/// ```rust
+/// # use hex_literal::hex;
+/// # use hopr_crypto_types::prelude::*;
+/// # use k256::ecdsa::VerifyingKey;
+///
+/// const PRIVATE_KEY: [u8; 32] = hex!("e17fe86ce6e99f4806715b0c9412f8dad89334bf07f72d5834207a9d8f19d7f8");
+///
+/// // compressed public keys start with `0x02` or `0x03``, depening on sign of y-component
+/// const COMPRESSED: [u8; 33] = hex!("021464586aeaea0eb5736884ca1bf42d165fc8e2243b1d917130fb9e321d7a93b8");
+///
+/// // full public key without prefix
+/// const UNCOMPRESSED_PLAIN: [u8; 64] = hex!("1464586aeaea0eb5736884ca1bf42d165fc8e2243b1d917130fb9e321d7a93b8fb0699d4f177f9c84712f6d7c5f6b7f4f6916116047fa25c79ef806fc6c9523e");
+///
+/// // uncompressed public keys use `0x04` prefix
+/// const UNCOMPRESSED: [u8; 65] = hex!("041464586aeaea0eb5736884ca1bf42d165fc8e2243b1d917130fb9e321d7a93b8fb0699d4f177f9c84712f6d7c5f6b7f4f6916116047fa25c79ef806fc6c9523e");
+///
+/// let from_privkey = PublicKey::from_privkey(&PRIVATE_KEY).unwrap();
+/// let from_compressed = PublicKey::from_bytes(&COMPRESSED).unwrap();
+/// let from_uncompressed_plain = PublicKey::from_bytes(&UNCOMPRESSED_PLAIN).unwrap();
+/// let from_uncompressed = PublicKey::from_bytes(&UNCOMPRESSED).unwrap();
+///
+/// assert_eq!(from_privkey, from_uncompressed);
+/// assert_eq!(from_compressed, from_uncompressed_plain);
+/// assert_eq!(from_uncompressed_plain, from_uncompressed);
+///
+/// // also works from a signed Ethereum transaction
+/// const TX_HASH: [u8; 32] = hex!("eff80b9f035b1d369c6a60f362ac7c8b8c3b61b76d151d1be535145ccaa3e83e");
+///
+/// const R: [u8; 32] = hex!("c8048d137fbb10ddffa1e4ba5141c300fcd19e4fb7d0a4354ca62a7694e46f9b");
+/// const S: [u8; 32] = hex!("5bb43e23d8b430f17ba3649e38b5a94d02815f0bcfaaf171800c52d4794c3136");
+/// const V: u8 = 1u8;
+///
+/// let mut r_and_s = Vec::<u8>::with_capacity(64);
+/// r_and_s.extend_from_slice(&R);
+/// r_and_s.extend_from_slice(&S);
+///
+/// let sig = Signature::new(&r_and_s, V);
+///
+/// let from_transaction_signature = PublicKey::from_signature_hash(&TX_HASH, &sig).unwrap();
+///
+/// assert_eq!(from_uncompressed, from_transaction_signature);
```

Comment:
The code is built, but not executed.

Reply:
are you sure? I tried it and I was even able to produce stdout output